Sources/TSFFutures/BatchingFutureOperationQueue.swift

```diff
@@ -51,32 +81,52 @@ public struct LLBBatchingFutureOperationQueue {
     ///    - group:     Threads capable of running futures.
     ///    - maxConcurrentOperationCount:
     ///                 Operations to execute in parallel.
-    @inlinable
-    public init(name: String, group: LLBFuturesDispatchGroup, maxConcurrentOperationCount maxOpCount: Int, qualityOfService: QualityOfService = .default) {
+    public convenience init(name: String, group: LLBFuturesDispatchGroup, maxConcurrentOperationCount maxOpCount: Int) {
+        self.init(name: name, group: group, maxConcurrentOperationCount: maxOpCount, dispatchQoS: .default)
+    }
+
+    public init(name: String, group: LLBFuturesDispatchGroup, maxConcurrentOperationCount maxOpCnt: Int, dispatchQoS: DispatchQoS) {
         self.group = group
-        self.operationQueue = OperationQueue(tsf_withName: name, maxConcurrentOperationCount: maxOpCount)
-        self.operationQueue.qualityOfService = qualityOfService
+        self.dispatchQueue = DispatchQueue(label: name, qos: dispatchQoS, attributes: .concurrent)
```

Comment:
hmm, `.concurrent` DispatchQueues are not recommended for a number of years now. Additionally, they're really quite slow to spawn new threads on Linux (can introduce a latency of up to 100 ms).

I'm not quite sure why we need the `DispatchQueue` at all?

You could for example use the [`TokenBucket` implementation](https://github.pie.apple.com/omega/libCAS/blob/a0a0c7c48aa9e9cb5da4dd7137c90866772198ff/Sources/CASTree/TokenBucket.swift) and just implement this class by doing

```
class LLBBatchingFutureOperationQueue {
    private let concurrencyLimiter = TokenBucket(initialShares: numberOfConcurrentOps)

   public func execute<T>(_ body: @escaping () throws -> T) -> LLBFuture<T> {
       let eventLoop = self.group.any()
       let promise = eventLoop.makePromise(of: T.self)
       return self.concurrencyLimiter.withReplenishableShares(numberOfShares: 1, eventLoop: eventLoop) { eventLoop in
           let offloadQ = DispatchQueue(label: "com.apple...offloadQ") // to make sure we don't block the EventLoop if `body` is blocking.
           offloadQ.async {
               do {
                    promise.succeed(try body())
               } catch {
                    promise.fail(error)
               }
           return promise.futureResult
       }
    }

   public func execute<T>(_ body: @escaping () -> LLBFuture<T>) -> LLBFuture<T> {
       return self.concurrencyLimiter.withReplenishableShares(numberOfShares: 1, eventLoop: eventLoop) { eventLoop in
            return body()
        }
    }
}
```